PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/component/ApplicationInsightsPanel.java

```diff
@@ -0,0 +1,188 @@
+    private ApplicationInsightsWrapper newInsightsWrapper;
+
+    public ApplicationInsightsPanel() {
+        cbInsights.setRenderer(new SimpleListCellRenderer() {
+            @Override
+            public void customize(@NotNull final JList list,
+                                  final Object o,
+                                  final int i,
+                                  final boolean b,
+                                  final boolean b1) {
+                if (o instanceof ApplicationInsightsWrapper) {
+                    final ApplicationInsightsWrapper insights = (ApplicationInsightsWrapper) o;
+                    setText(insights.isNewCreated ?
+                            String.format(AzureFunctionsConstants.NEW_CREATED_RESOURCE, insights.name) :
+                            String.format("%s (Resource Group: %s)", insights.name, insights.resourceGroup));
+                } else if (o instanceof String) {
+                    setText((String) o);
+                }
+            }
+        });
+
+        cbInsights.addPopupMenuListener(new PopupMenuListenerAdapter() {
+            @Override
+            public void popupMenuWillBecomeInvisible(final PopupMenuEvent e) {
+                onSelectApplicationInsights();
+            }
+        });
+
+        newInsightsWrapper = ApplicationInsightsWrapper.wrapperNewInsightsInstance();
+    }
+
+    public void loadApplicationInsights(String subscriptionId) {
+        this.subscriptionId = subscriptionId;
+        if (subscription != null && !subscription.isUnsubscribed()) {
+            subscription.unsubscribe();
+        }
+        beforeLoadApplicationInsights();
+        subscription = Observable
+                .fromCallable(() -> AzureSDKManager.getInsightsResources(subscriptionId))
+                .subscribeOn(Schedulers.newThread())
+                .subscribe(this::fillApplicationInsights);
+    }
+
+    public void changeDefaultApplicationInsightsName(String name) {
+        newInsightsWrapper.setName(name);
+        cbInsights.repaint();
+    }
+
+    public boolean isCreateNewInsights() {
+        return selectWrapper == null ? false : selectWrapper.isNewCreated;
+    }
+
+    public String getApplicationInsightsInstrumentKey() {
+        return selectWrapper == null ? null : selectWrapper.instrumentKey;
+    }
+
+    public String getNewApplicationInsightsName() {
+        return selectWrapper == null ? null : selectWrapper.name;
+    }
+
+    private void onSelectApplicationInsights() {
+        final Object selectedObject = cbInsights.getSelectedItem();
+        if (CREATE_NEW_APPLICATION_INSIGHTS.equals(selectedObject)) {
+            ApplicationManager.getApplication().invokeLater(this::createApplicationInsights);
+        } else if (selectedObject instanceof ApplicationInsightsWrapper) {
+            selectWrapper = (ApplicationInsightsWrapper) selectedObject;
+        }
+    }
+
+    private void createApplicationInsights() {
```

Comment:
createApplicationInsights is a bad name since this method will not actually create the AI in azure.